### PR TITLE
List samples with transfer dialog

### LINKED ISSUE #468: Add list of samples to transfer dialog

### PR DESCRIPTION
**Pull request by ksierks**
## What does this PR do and why?
Moved the checkboxes into their own column within the group &  project samples tables. 
Added a list of selected samples to the transfer dialog.
Fixes #468.

## Screenshots or screen recordings
Screenshot from 2024-03-20 15-31-12
image

## How to set up and validate locally
1. Login to irida-next.
2. Navigate to a project samples page that the user is a maintainer or owner. 
3. Select some samples and click the `Transfer samples` button.
4. Verify the selected samples are listed within the dialog.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

File: config/locales/en.yml [hunk 1]
```diff
@@ -956,6 +956,7 @@ en:
       transfers:
         dialog:
           title: Transfer Samples
+          description: "The following files were selected to be transferred:"
```
Comment (joshsadam) [inline config/locales/en.yml:959]: **nitpick:** This should probably read `The following samples were selected to be transfered`

File: app/javascript/controllers/list_samples_controller.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+// creates a table listing all the selected samples
+export default class extends Controller {
+  static targets = ["field"];
+
+  #td_classes = ["bg-slate-50", "dark:bg-slate-900", "left-0", "left-12", "sticky"];
+
+  connect() {
+    const table = document.getElementById("samples-table");
+    const body = table.getElementsByTagName("tbody")[0];
+    const newTable = document.createElement("table");
```
Comment (joshsadam) [inline app/javascript/controllers/list_samples_controller.js:12]: **suggestion:** This table will need table headers.  Try having the table already in the dialog with the headers and have the body as a target for this controller.

**Review by joshsadam — CHANGES_REQUESTED** [review]
Doing the same thing as files (cloning the first two columns) won't work here.  This is what I tried:
1. I have a project of 40 (anything >20 will work)
2. Select all samples in the project
3. Open then transfer dialog, only 20 samples are displayed ☹️

**Comment by ksierks** [comment]
redoing in another pr

**Pull request closed without merging**